tests/v2/validation/rbac/globalrolesv2/globalroles_v2_test.go

```diff
@@ -0,0 +1,501 @@
+	"github.com/rancher/rancher/tests/framework/extensions/provisioning"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type GlobalRolesV2TestSuite struct {
+	suite.Suite
+	client       *rancher.Client
+	session      *session.Session
+	clusterCount int
+	user         *management.User
+	grbName      string
+	grName       string
+}
+
+func (gr *GlobalRolesV2TestSuite) TearDownSuite() {
+	gr.session.Cleanup()
+}
+
+func (gr *GlobalRolesV2TestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	gr.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(gr.T(), err)
+
+	gr.client = client
+}
+
+// Private helper function to create a Global Role with inheritedClusterRole and validate if GRBs, CRTBs, CRBs and RBs are created successfully.
+func (gr *GlobalRolesV2TestSuite) createAndTestGlobalRole(inheritedRoles []string) {
+	log.Info("Create a global role with inheritedClusterRoles.")
+	globalRole.Name = namegen.AppendRandomString("testgr")
+	globalRole.InheritedClusterRoles = inheritedRoles
+	createdGlobalRole, err := rbac.CreateGlobalRole(gr.client, localcluster, &globalRole)
+	require.NoError(gr.T(), err)
+	gr.grName = createdGlobalRole.Name
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(gr.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(gr.T(), err)
+	gr.user = createdUser
+
+	log.Info("Verify that the global role binding is created for the user.")
+	grbOwner, err := getGlobalRoleBindingForUser(gr.client, createdUser.ID)
+	require.NoError(gr.T(), err)
+	require.NotEmpty(gr.T(), grbOwner, "Global Role Binding not found for the user")
+	gr.grbName = grbOwner
+
+	log.Info("Verify that the cluster role template bindings are created for the downstream clusters.")
+	clusterNames, err := clusters.ListDownstreamClusters(gr.client)
+	require.NoError(gr.T(), err)
+	gr.clusterCount = len(clusterNames)
+	expectedCrtbCount := gr.clusterCount * len(inheritedRoles)
+	crtbs, err := listClusterRoleTemplateBindingsForGlobalRoleBinding(gr.client, grbOwner, expectedCrtbCount)
+	require.NoError(gr.T(), err)
+	actualCrtbCount := len(crtbs.Items)
+	require.Equal(gr.T(), expectedCrtbCount, actualCrtbCount, "Unexpected number of ClusterRoleTemplateBindings: Expected %d, Actual %d", expectedCrtbCount, actualCrtbCount)
+
+	log.Info("Verify that the cluster role bindings are created for the downstream cluster.")
+	expectedCrbCount := expectedCrtbCount
+	crbs, err := getCRBsForCRTBs(gr.client, crtbs)
+	require.NoError(gr.T(), err)
+	actualCrbCount := len(crbs.Items)
+	require.Equal(gr.T(), expectedCrbCount, actualCrbCount, "Unexpected number of ClusterRoleBindings: Expected %d, Actual %d", expectedCrbCount, actualCrbCount)
+
+	log.Info("Verify that the role bindings are created for the downstream cluster.")
+	expectedRbCount := expectedCrtbCount
```

Comment:
Similar to CRBs, the expectedRBCount may vary with different roles.

Reply:
Based on our discussion, we will leave this code for CRBs and RBs as is.